dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_memory.cc

```diff
@@ -208,17 +208,39 @@ static int DeviceNew(void **ptr, size_t size) {
     *ptr = nullptr;
     return cudaSuccess;
   }
-  *ptr = GetBuffer(std::this_thread::get_id(), AllocType::GPU, size);
-  return *ptr != nullptr ? cudaSuccess : cudaErrorMemoryAllocation;
+  // this function should not throw, but return a proper result
+  try {
+    *ptr = GetBuffer(std::this_thread::get_id(), AllocType::GPU, size);
+    return *ptr != nullptr ? cudaSuccess : cudaErrorMemoryAllocation;
+  } catch (const std::bad_alloc &) {
```

Comment:
```suggestion
  } catch (const CUDABadAlloc &) {
```

Reply:
Not sure about that one, though.

Reply:
Allocator throws `std::bad_alloc` https://github.com/NVIDIA/DALI/blob/master/dali/kernels/alloc.h#L43. Although `Deallocate` uses CUDA_DTOR_CALL and that one can throw CUDABadAlloc . I guess we wanted to translate all nullptrs (Even if requested size is 0) as CUDABadAlloc.